core/modules/image/image.tokens.inc

```diff
@@ -0,0 +1,327 @@
+<?php
+/**
+ * @file
+ * Builds placeholder replacement tokens for image styles.
+ */
+
+/**
+ * Implements hook_token_info().
+ */
+function image_token_info() {
+  $styles = image_styles();
+  $styles_tokens = array();
+  $image_style = array();
+
+  foreach ($styles as $style => $desc) {
+    $styles_tokens[$style] = array(
+      'name' => $style,
+      'description' => t("@s image style", array("@s" => $style)),
+      'type' => 'image-style',
+    );
+    foreach (_imagegecache_token_image_attributes() as $attribute => $none) {
+      $image_style[$attribute] = array(
+        'name' => $attribute,
+        'description' => t("@s image style attribute: @a", array("@s" => $style, "@a" => $attribute)),
+      );
+    }
+  }
+
+  return array(
+    'types' => array(
+      'image-field' => array(
+        'name' => 'Image Field',
+        'description' => 'Image Field',
+        'needs-data' => 'image-field',
+      ),
+      'image-style' => array(
+        'name' => 'Image Style',
+        'description' => 'Image Style',
```

Comment:
Not sure why capitalization is being used here:

```suggestion
      'image-field' => array(
        'name' => 'Image field',
        'description' => 'Image field',
        'needs-data' => 'image-field',
      ),
      'image-style' => array(
        'name' => 'Image style',
        'description' => 'Image style',
```

Reply:
fixed